src/resolve-remote.ts

```diff
@@ -0,0 +1,111 @@
+import cacache from 'cacache';
+import PQueue from 'p-queue';
+import validatePackageName from 'validate-npm-package-name';
+import {SnowpackConfig} from './config.js';
+import {InstallTarget} from './scan-imports.js';
+import {fetchCDNResource, ImportMap, PIKA_CDN, RESOURCE_CACHE, HAS_CDN_HASH_REGEX} from './util.js';
+
+/**
+ * Given an install specifier, attempt to resolve it from the CDN.
+ * If no lockfile exists or if the entry is not found in the lockfile, attempt to resolve
+ * it from the CDN directly. Otherwise, use the URL found in the lockfile and attempt to
+ * check the local cache first.
+ *
+ * All resolved URLs are populated into the local cache, where our internal Rollup engine
+ * will load them from when it installs your dependencies to disk.
+ */
+async function resolveDependency(
+  installSpecifier: string,
+  packageSemver: string,
+  lockfile: ImportMap | null,
+): Promise<null | string> {
+  // Right now, the CDN is only for top-level JS packages. The CDN doesn't support CSS,
+  // non-JS assets, and has limited support for deep package imports. Snowpack
+  // will automatically fall-back any failed/not-found assets from local
+  // node_modules/ instead.
+  if (!validatePackageName(installSpecifier).validForNewPackages) {
+    return null;
+  }
+
+  // Grab the installUrl from our lockfile if it exists, otherwise resolve it yourself.
+  let installUrl: string;
+  if (lockfile && lockfile.imports[installSpecifier]) {
+    installUrl = lockfile.imports[installSpecifier];
+  } else {
+    if (packageSemver === 'latest') {
+      console.warn(
+        `warn(${installSpecifier}): Not found in "dependencies". Using latest package version...`,
+      );
+    }
+    if (packageSemver.startsWith('npm:@reactesm') || packageSemver.startsWith('npm:@pika/react')) {
+      throw new Error(
+        `React workarounds no longer needed in --source=pika mode. Revert to the official React & React-DOM packages.`,
```

Comment:
Is it worth also mentioning this in the docs? Probably with the react guide?

Reply:
great call